pkg/reconciler/taskrun/taskrun.go

```diff
@@ -527,6 +530,16 @@ func (c *Reconciler) reconcile(ctx context.Context, tr *v1beta1.TaskRun, rtr *re
 	}
 
 	if podconvert.SidecarsReady(pod.Status) {
+		if spireEnabled {
+			// TTL for the entry is in seconds
+			ttl := time.Duration(config.FromContextOrDefaults(ctx).Defaults.DefaultTimeoutMinutes) * time.Minute
+			if err = c.SpireClient.CreateEntries(ctx, tr, pod, ttl); err != nil {
+				logger.Errorf("Failed to create workload SPIFFE entry for taskrun %v: %v", tr.Name, err)
+				return err
+			}
+			logger.Infof("Created SPIFFE workload entry for %v/%v", tr.Namespace, tr.Name)
+		}
```

Comment:
I wonder if this can be tested via unit tests. Coverage is missing for this. Maybe it can only be done via e2e tests?

Reply:
Creating the entries cannot be checked as this is an interaction with the spire server directly. We do have some mocked spire tests in the spire package but this has to be done via e2e test.